hibernate-core/src/main/java/org/hibernate/type/ComponentType.java

```diff
@@ -176,9 +176,7 @@ public boolean isEqual(final Object x, final Object y) throws HibernateException
 		if ( x == y ) {
 			return true;
 		}
-		if ( x == null || y == null ) {
```

Comment:
Done. It was a new conflict in AvailableSettings as there were new changes since my last rebase.